code/modules/mob/living/carbon/carbon.dm

```diff
@@ -269,14 +269,22 @@ mob/living
 						)
 			// BEGIN HUGCODE - N3X
 			else
-				if (istype(src,/mob/living/carbon/human) && src:w_uniform)
-					var/mob/living/carbon/human/H = src
-					H.w_uniform.add_fingerprint(M)
-				playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
-				M.visible_message( \
-					"\blue [M] gives [src] a [pick("hug","warm embrace")].", \
-					"\blue You hug [src].", \
+				if(M.zone_sel.selecting == "head")
+					playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+					M.visible_message( \
+					"<span class='notice'> [M] pats [src] on the head.</span>", \
+					"<span class='notice'> You pat [src] on the head.</span>", \
 					)
+					src << "Getting patted by [M]-senpai fills you with determination"
+				else
+					playsound(get_turf(src), 'sound/weapons/thudswoosh.ogg', 50, 1, -1)
+					M.visible_message( \
+					"<span class='notice'> [M] gives [src] a [pick("hug","warm embrace")].</span>", \
+					"<span class='notice'> You hug [src].</span>", \
+					)
+					if (istype(src,/mob/living/carbon/human) && src:w_uniform)
```

Comment:
> I know that this was already here, but since you're touching it anyway, colon usage is bad. Typecast and use a dot operator.

Git is being weird and thought this line was changed.